flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -63,11 +64,14 @@
  * The yarn implementation of the resource manager. Used when the system is started
  * via the resource framework YARN.
  */
-public class YarnResourceManager extends ResourceManager<ResourceID> implements AMRMClientAsync.CallbackHandler {
+public class YarnResourceManager extends ResourceManager<YarnWorkerNode> implements AMRMClientAsync.CallbackHandler {
 
 	/** The process environment variables. */
 	private final Map<String, String> env;
 
+	/** YARN container map. Package private for unit test purposes. */
+	final Map<String, YarnWorkerNode> workerNodeMap;
```

Comment:
Let's make it more visible that this map is a concurrent map by setting the type to `CocnurrentHashMap`.